File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_accountpagedresponsemodel.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import codat_datacontracts_datasets_accountpagedresponselinksmodel as shared_codat_datacontracts_datasets_accountpagedresponselinksmodel
from ..shared import codat_datacontracts_datasets_account as shared_codat_datacontracts_datasets_account


@dataclass_json
@dataclasses.dataclass
class CodatDataContractsDatasetsAccountPagedResponseModel:
    r"""CodatDataContractsDatasetsAccountPagedResponseModel
    Used to represent what can be returned by an endpoint that supports paging.
    Usable with the [ProducesResponseType] attribute on a controller action.
    """
    
    links: Optional[shared_codat_datacontracts_datasets_accountpagedresponselinksmodel.CodatDataContractsDatasetsAccountPagedResponseLinksModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('_links') }})
    page_number: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageNumber') }})
    page_size: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('pageSize') }})
    results: Optional[list[shared_codat_datacontracts_datasets_account.CodatDataContractsDatasetsAccount]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('results') }})
    total_results: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('totalResults') }})
    

File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsaccountpagedresponsemodel.go
package shared

// CodatDataContractsDatasetsAccountPagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type CodatDataContractsDatasetsAccountPagedResponseModel struct {
	Links        *CodatDataContractsDatasetsAccountPagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                                      `json:"pageNumber,omitempty"`
	PageSize     *int                                                      `json:"pageSize,omitempty"`
	Results      []CodatDataContractsDatasetsAccount                       `json:"results,omitempty"`
	TotalResults *int                                                      `json:"totalResults,omitempty"`
}
